src/main/java/com/mariamkatamashvlii/gym/serviceImpl/TraineeServiceImpl.java

```diff
@@ -0,0 +1,178 @@
+        }
+    }
+
+    @Override
+    public void updateTrainerList(String username, String password, List<Trainer> trainerList) {
+        if (userService.checkCredentials(username, password)) {
+            User user = userRepo.select(username);
+            Trainee trainee = user.getTrainee();
+            trainee.setTrainerList(trainerList);
+            traineeRepo.update(trainee);
+            userRepo.update(user);
+            logger.info("Updated trainer list for trainee with username - {}", trainee.getUser().getUsername());
+        }
+    }
+
+    @Override
+    public List<Trainee> findAll() {
+        logger.info("Returning all trainees");
+        return traineeRepo.findAll();
+    }
+
+    @Override
+    public void createTraineeProfile(Date dob, String address, long userId) {
+        Trainee trainee = new Trainee(dob, address);
+        User user = userRepo.select(userId);
+        trainee.setUser(user);
+        create(trainee);
+        logger.info("Created trainee profile with id {}", trainee.getTraineeId());
+    }
+
+    @Override
+    public List<Training> geTrainingList(String username, String password, Date fromDate, Date toDate, String trainerName, TrainingType trainingType) {
+        List<Training> trainingList = new ArrayList<>();
+        if (userService.checkCredentials(username, password)) {
+            Trainee trainee = traineeRepo.select(username);
+            Set<Training> trainings = trainee.getTrainingSet();
+            for (Training t : trainings) {
+                if (isBetween(t.getTrainingDate(), fromDate, toDate) &&
+                        t.getTrainer().getUser().getFirstName().equals(trainerName) &&
+                        t.getTrainingType().getTrainingTypeName().equals(trainingType.getTrainingTypeName())) {
+                    trainingList.add(t);
+                }
+            }
+        }
+        logger.info("Returning training list for trainee {}", username);
+        return trainingList;
+    }
+
+    @Override
+    public List<Trainer> getNotAssignedTrainerList(String username, String password) {
+        List<Trainer> notAssignedTrainerList = new ArrayList<>();
+        if (userService.checkCredentials(username, password)) {
+            Trainee trainee = traineeRepo.select(username);
+            List<Trainer> traineeTrainers = trainee.getTrainerList();
+            List<Trainer> allTrainers = trainerRepo.findAll();
+            for (Trainer t : allTrainers) {
+                if (!traineeTrainers.contains(t)) {
+                    notAssignedTrainerList.add(t);
+                }
+            }
+        }
+        logger.info("Returning not assigned trainer list for trainee {}", username);
+        return notAssignedTrainerList;
+    }
+
+    public boolean isBetween(Date trainingdate, Date fromDate, Date toDate) {
+        return trainingdate.compareTo(fromDate) >= 0 && trainingdate.compareTo(toDate) <= 0;
+    }
```

Comment:
Ususally, we use LocalDate/LocalDateTime for date and time.
They have very convinient method for processing.